source/jormungandr/jormungandr/street_network/street_network.py

```diff
@@ -38,16 +38,23 @@
 # http://stackoverflow.com/a/38668373/1614576
 ABC = abc.ABCMeta(str("ABC"), (object,), {})
 
+# Regarding to the type of direct path, some special treatments may be done in connector
+class DirectPathType:
+    DIRECT_NO_PT = 0
+    BEGINNING_FALLBACK = 1
+    ENDING_FALLBACK = 2
 
 class AbstractStreetNetworkService(ABC):
     @abc.abstractmethod
     def get_street_network_routing_matrix(self, origins, destinations, street_network_mode, max_duration, request, **kwargs):
         pass
 
     @abc.abstractmethod
-    def direct_path(self, mode, pt_object_origin, pt_object_destination, fallback_extremity, request):
+    def direct_path(self, mode, pt_object_origin, pt_object_destination, fallback_extremity, request, direct_path_type):
```

Comment:
You probably have to do a grep for `direct_path(` in connectors: geovelo should be affected by this change too.
tests also (kraken and geovelo?)